src/DSharp.Mscorlib/Collections/Generic/List.cs

```diff
@@ -10,127 +11,70 @@ namespace System.Collections.Generic
+        public extern int IndexOf(object value);
 
         public extern int IndexOf(T item);
 
-        public extern int IndexOf(T item, int startIndex);
-
-        public extern void Insert(int index, T item);
-
-        public extern void InsertRange(int index, params T[] items);
-
-        public extern string Join();
-
-        public extern string Join(string delimiter);
-
-        public extern int LastIndexOf(object item);
-
-        public extern int LastIndexOf(object item, int fromIndex);
-
-        public extern List<TTarget> Map<TTarget>(ListMapCallback<T, TTarget> mapCallback);
-
-        public extern List<TTarget> Map<TTarget>(ListItemMapCallback<T, TTarget> mapItemCallback);
-
-        public extern static List<T> Parse(string s);
-
-        public extern TReduced Reduce<TReduced>(ListReduceCallback<TReduced, T> callback);
-
-        public extern TReduced Reduce<TReduced>(ListReduceCallback<TReduced, T> callback, TReduced initialValue);
-
-        public extern TReduced Reduce<TReduced>(ListItemReduceCallback<TReduced, T> callback);
-
-        public extern TReduced Reduce<TReduced>(ListItemReduceCallback<TReduced, T> callback, TReduced initialValue);
-
-        public extern TReduced ReduceRight<TReduced>(ListReduceCallback<TReduced, T> callback);
-
-        public extern TReduced ReduceRight<TReduced>(ListReduceCallback<TReduced, T> callback, TReduced initialValue);
-
-        public extern TReduced ReduceRight<TReduced>(ListItemReduceCallback<TReduced, T> callback);
-
-        public extern TReduced ReduceRight<TReduced>(ListItemReduceCallback<TReduced, T> callback, TReduced initialValue);
+        [DSharpScriptMemberName("remove")]
+        public extern void Remove(object value);
 
         [DSharpScriptMemberName("remove")]
         public extern bool Remove(T item);
 
         public extern void RemoveAt(int index);
 
-        public extern List<T> RemoveRange(int index, int count);
-
-        public extern void Reverse();
-
-        public extern List<T> Slice(int start);
-
-        public extern List<T> Slice(int start, int end);
-
-        public extern bool Some(ListFilterCallback<T> filterCallback);
-
-        public extern bool Some(ListItemFilterCallback<T> itemFilterCallback);
+        public extern IEnumerator GetEnumerator();
 
-        public extern void Sort();
+        extern IEnumerator<T> IEnumerable<T>.GetEnumerator();
 
-        public extern void Sort(CompareCallback<T> compareCallback);
-
-        public extern void Splice(int start, int deleteCount);
+        public extern void Insert(int index, T item);
 
-        public extern void Splice(int start, int deleteCount, params T[] itemsToInsert);
+        public extern void ForEach(ListCallback<T> callback);
 
-        public extern void Unshift(params T[] items);
+        public extern void ForEach(ListItemCallback<T> itemCallback);
 
         [ScriptSkip]
         public extern T[] ToArray();
 
+        [ScriptSkip]
```

Comment:
I'd actually be super interested if we could create a frozen wrapped version of the array which is actually Readonly here. 
Could have static a method called here instead which does that 🤔

Reply:
`Object.freeze` might be used for that, but we don't really want to do that. I would use ReadOnly just for type safety